Added support for `useCurrentOnUpdate` modifier

promatik opened the pull request:
Since [v8.36.0](https://github.com/laravel/framework/blob/b09394ed478551d01f5a6ce3111e4d140041bc82/CHANGELOG-8.x.md#v8360-2021-04-06), Laravel supports the `useCurrentOnUpdate` modifier – https://github.com/laravel/framework/pull/36817
This PR adds it to the list of `$modifiers`.

jasonmccreary commented:
@promatik, could you add this to a draft file for one of the migrations (or make a new one) so we have coverage.

promatik commented:
Sure @jasonmccreary, I added it to `ModelLexerTest.php`, there was a test using `useCurrent` on a `saved_at` column. I added another column named `updated_at` with `useCurrentOnUpdate`.

promatik commented:
End up adding it to the `drafts/post.yaml` too.
